### PR TITLE
Explicitly disallow bad versions of pytest when using test runner

### LINKED ISSUE #8177: Tests are not picked up for pytest >=3.7 on LTS
For some yet unknown reason the tests are not picked up by the test collection only the rst files from the narrative documentation on the LTS branch when using pytest 3.7+

I think at least understanding, preferable fixing, this is critical for the release.

```
================================================= test session starts ==================================================
platform darwin -- Python 3.7.0, pytest-3.7.4, py-1.6.0, pluggy-0.7.1

Running tests with Astropy version 2.0.10.dev19739.
Running tests in astropy docs.

Date: 2018-11-24T13:44:00

Platform: Darwin-17.7.0-x86_64-i386-64bit

Executable: /usr/local/opt/python/bin/python3.7

Full Python Version: 
3.7.0 (default, Sep 18 2018, 18:47:22) 
[Clang 9.1.0 (clang-902.0.39.2)]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Numpy: 1.15.4
Scipy: 1.1.0
Matplotlib: 3.0.2
h5py: 2.8.0
Pandas: 0.23.4
Cython: 0.28.5
astropy_helpers: 3.1rc1
Using Astropy options: remote_data: none.

rootdir: /private/var/folders/dc/hsm7tqpx2d57n7vb3k1l81xw0000gq/T/astropy-test-l155hvv3/lib/python3.7/site-packages, inifile: setup.cfg
plugins: cov-2.6.0, arraydiff-0.2
collected 168 items                                                                                                    

docs/changelog.rst .                                                                                             [  0%]
docs/credits.rst .                                                                                               [  1%]
docs/getting_started.rst .                                                                                       [  1%]
docs/importing_astropy.rst .                                                                                     [  2%]
...
...
docs/whatsnew/2.0.rst .                                                                                          [ 99%]
docs/whatsnew/index.rst .                                                                                        [100%]

============================================== 168 passed in 8.68 seconds ==============================================
```

cc @dan @eteq @astrofrog

### PR DESCRIPTION
**Pull request by drdavella**
This partially resolves #8177.

### CODE REVIEW

**Comment by astrofrog** [comment]
Great, and I think we need https://github.com/astropy/astropy/pull/8187 too so that when it gets auto-installed, a compatible version is used.

File: astropy/tests/runner.py [hunk 1]
```diff
@@ -183,6 +183,15 @@ def run_tests(self, **kwargs):
 
         # Don't import pytest until it's actually needed to run the tests
         import pytest
+        if pytest.__version__ > '3.7' and pytest.__version__ < '4':
```
Comment (bsipocz) [inline astropy/tests/runner.py]: actually it's incompatible with <4 as well, due to the deprecation warnings that turned into exceptions in that version.

Comment (astrofrog) [reply astropy/tests/runner.py]: Also I don't think ``pytest.__version__`` is safe as once pytest reaches version 10.x this check will fail. If they go like setuptools, version 10.x is not that unlikely.

Comment (pllim) [reply astropy/tests/runner.py]: Given how often they are releasing lately, I am not surprised if they hit 10.x in the near future.

Comment (bsipocz) [reply astropy/tests/runner.py]: wouldn't LooseVersion solve this issue?

File: astropy/tests/runner.py [hunk 2]
```diff
@@ -183,6 +183,18 @@ def run_tests(self, **kwargs):
 
         # Don't import pytest until it's actually needed to run the tests
         import pytest
+        from distutils.version import LooseVersion
+
+        pytest_version = LooseVersion(pytest.__version__)
+        if pytest_version > '3.7':
```
Comment (astrofrog) [inline astropy/tests/runner.py]: ``>=``

Comment (drdavella) [reply astropy/tests/runner.py]: Gah, thanks.

**Comment by astrofrog** [comment]
@drdavella - can you remove the changes to astropy-helpers? 😁

**Comment by bsipocz** [comment]
If it were me, I would ban the usage of `git add -A` 😄 :trollface:

**Comment by drdavella** [comment]
If it were me, I would ban the usage of git submodules 😁 :trollface:.

But it's mostly karma, because I have gotten on enough other people about this.

**Comment by bsipocz** [comment]
But not using `git add -A` would solve all kind of issues not just submodule related ones ;)
But I agree, and came on board with the whole idea of getting rid of the helpers as a submodule.

**Comment by bsipocz** [comment]
Please update the travis matrix to remove all too new pytest versions, then I think this can be merged.

(circleCI issue has been fixed since, so doing a rebase would get rid of it)

File: astropy/tests/runner.py [hunk 3]
```diff
@@ -183,6 +183,18 @@ def run_tests(self, **kwargs):
 
         # Don't import pytest until it's actually needed to run the tests
         import pytest
+        from distutils.version import LooseVersion
+
+        pytest_version = LooseVersion(pytest.__version__)
+        if pytest_version >= '3.7':
```
Comment (bsipocz) [inline astropy/tests/runner.py]:
```suggestion
        if pytest_version >= LooseVersion('3.7'):
```

Comment (bsipocz) [reply astropy/tests/runner.py]: weird, now I see the buttons to add the suggestion to a commit, but it definitely wasn't there before even after I assigned myself (thus I sent a feedback for the beta feature).

**Comment by bsipocz** [comment]
the last commit got rid of the failing pytest setting in travis, so I'm merging this now.

Thanks @drdavella!

**Pull request merged**